### DISCUSSION TITLE
c-index in randomforestSRC competing risk

### DISCUSSION THREAD

**Discussion by RUID818** [kogalur/randomForestSRC#419] [Q&A]
Hi, 
I am trying to build a random survival forest with four competing risks. I was just wondering how to get the correct c-index. My understanding is that the get.cindex doesn't work for competing risks. I was wondering if the OOB requested performance error  works as a c-index. In random survival forest without competing risk, I can always get exactly the same value from the OOB performance error as from the get.cindex function.  As I attached below, the first competing risk(event=51) is the risk of interest. I am wondering if the value of 0.28451287 is the correct c-index we can use. 
Thank you so much,
Rui

 Sample size: 801
                    Number of events: 51, 61, 38, 93
                     Number of trees: 10000
           Forest terminal node size: 15
       Average no. of terminal nodes: 44.7012
No. of variables tried at each split: 8
              Total no. of variables: 56
       Resampling used to grow trees: swor
    Resample size used to grow trees: 506
                            Analysis: RSF
                              Family: surv-CR
                      Splitting rule: logrank *random*
       Number of random split points: 10
   (OOB) Requested performance error: **0.28451287**, 0.26050922, 0.43872033, 0.329127

**Comment by ishwaran** [comment]
Yes, the OOB requested performance error works and returns the C-index for competing risk settings.  The ` get.cindex` function is only used for right-censored survival data (non-competing risk).

Here's a competing risk illustration:

```
data(follic, package = "randomForestSRC")
follic.obj <- rfsrc(Surv(time, status) ~ ., follic, nsplit = 3, ntree = 100)
> follic.obj
                         Sample size: 541
                    Number of events: 272, 76
                     Number of trees: 100
           Forest terminal node size: 15
       Average no. of terminal nodes: 24.38
No. of variables tried at each split: 3
              Total no. of variables: 5
       Resampling used to grow trees: swor
    Resample size used to grow trees: 342
                            Analysis: RSF
                              Family: surv-CR
                      Splitting rule: logrankCR *random*
       Number of random split points: 3
   (OOB) Requested performance error: 0.40100754, 0.27999674

```

You can always pull the error rate from $err.rate for any survival family to confirm the right answer:

```
> tail(follic.obj$err.rate,1)
         event.1   event.2
[100,] 0.4010075 0.2799967
```

**Comment by RUID818** [comment]
Thank you so much!
